tests/underflow_svm/test_underflow_svm.c

```diff
@@ -0,0 +1,123 @@
+/********************************************************************************
+ * Copyright (c) 2016 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ********************************************************************************/
+
+
+// A test to make sure that programs with a buffer overflow complete and
+// cause the buffer overflow detector to find the overflow.
+#include "common_test_functions.h"
+
+const char *kernel_source = "\n"\
+"__kernel void test(__global uint *svm_buffer, uint len) {\n"\
+"    uint i = get_global_id(0);\n"\
+"    if (i < len) {\n"\
+"        *(__global uint *)(&svm_buffer[i] - 10) = i;\n"\
+"    }\n"\
+"}\n";
+
+int main(int argc, char** argv)
+{
+#ifdef CL_VERSION_2_0
+    cl_int cl_err;
+    uint32_t platform_to_use = 0;
+    uint32_t device_to_use = 0;
+    cl_device_type dev_type = CL_DEVICE_TYPE_DEFAULT;
+    uint64_t buffer_size = DEFAULT_BUFFER_SIZE;
+
+    // Check input options.
+    check_opts(argc, argv, "SVM with Overflow",
```

Comment:
Name of the file uses "underflow" -- if that's the terminology we're using then "overflow" should be changed in the code and printouts throughout this test.